### PR TITLE
Add login() function

### LINKED ISSUE #1: Finalize meditor ("Edit metadata") webshotting
there is already a stab https://github.com/dandi/dandi-api-webshots/blob/master/tools/make_webshots.py#L49 but we would need to have authenticated session (based on having `DANDI_API_KEY` env var provided) to be able to enter meditor for a dataset for a user which either has admin privileges or just owns a specific dandiset.

### PR DESCRIPTION
**Pull request by jwodder**
This PR adds a `login()` function to `make_webshots.py` for logging into https://gui-beta-dandiarchive-org.netlify.app.  I leave it to you how you want to manage the passing-in of credentials.

Closes #1.

### CODE REVIEW

**Comment by yarikoptic** [comment]
Please make it complete by taking username/password from `DANDI_USERNAME` and `DANDI_PASSWORD` env variables, then I will just merge and add those secrets for dandibot user.

**Comment by yarikoptic** [comment]
done -- added those two secrets and somehow that user already can enter/modify metadata although just registered.   So please add passing them into the script within the action itself as well.
@dchiquito is that expected that dandibot user can "Edit metadata"?

**Comment by jwodder** [comment]
@yarikoptic If you didn't already, could you log in the Dandibot user manually to https://gui-beta-dandiarchive-org.netlify.app/?  I think the script may be encountering a post-login prompt from GitHub that won't go away until it's been pressed once.

**Comment by yarikoptic** [comment]
Odd.. I did login. I will do again in new incognito widow in 5 minutes

File: .github/workflows/update.yaml [hunk 1]
```diff
@@ -34,6 +34,9 @@ jobs:
 
       - name: Run webshots script
         run: xvfb-run python tools/make_webshots.py
+        env:
+          DANDI_USERNAME: ${{ secrets.DANDI_USERNAME }}
+          DANDI_PASSWORD: ${{ secrets.DANDI_PASSWORD }}
```
Comment (yarikoptic) [inline .github/workflows/update.yaml:39]: iirc it includes `$` - I wonder if there is necessity to surround the value in `'` ?

**Comment by yarikoptic** [comment]
went to new incognito window but it (github?) remembered dandibot (well -- `??`) logged in.  When I clicked on `??` it had no API key.  I clicked again and API Key appeared... try now?  
but also see a comment about wild guess on possibly needing quoting the value?

File: .github/workflows/update.yaml (hunk 1 above)
Comment (jwodder) [reply .github/workflows/update.yaml:39]: This isn't being parsed by the shell, so that shouldn't be necessary.

**Comment by jwodder** [comment]
@yarikoptic In order to debug the failure, I made the script take a screenshot and uploaded it as an artifact.  It shows the GitHub login page with the message "Incorrect username or password."

File: tools/make_webshots.py [hunk 2]
```diff
@@ -38,8 +38,12 @@ def login(driver, username, password):
     password_field.send_keys(password)
     driver.find_elements_by_tag_name("form")[0].submit()
 
-    WebDriverWait(driver, 30).until(
-        EC.presence_of_element_located((By.CLASS_NAME, "v-avatar")))
+    try:
+        WebDriverWait(driver, 30).until(
+            EC.presence_of_element_located((By.CLASS_NAME, "v-avatar")))
```
Comment (yarikoptic) [inline tools/make_webshots.py:48]: FWIW this is what I see -- so there is indeed v-avatar div
image

**Comment by jwodder** [comment]
@yarikoptic I believe I've figured out the problem.  Because I didn't have write permission for this repo when I started this branch, I forked the repository and submitted this PR from that fork, but forks don't have access to workflow secrets in parent repositories.  I'm going to close this PR and submit a new one directly from the repository.

**Pull request closed without merging**